src/lightning_app/core/app.py

```diff
@@ -596,3 +603,11 @@ def _on_run_end(self):
         if os.getenv("LIGHTNING_DEBUG") == "2":
             del os.environ["LIGHTNING_DEBUG"]
             _console.setLevel(logging.INFO)
+
+    def _on_before_setup(self):
+        for w in self.works:
+            if w._start_before_setup:
+                parallel = w.parallel
+                w._parallel = True
+                w.start()
+                w._parallel = parallel
```

Comment:
I would move the `_parallel` hack to `start`, and give the possibility to call `start` manually from within `setup`. I would then eliminate this method.

Reply:
Moved to the runtime. This is required only for multiprocessing locally.